__tests__/app.test.js

```diff
@@ -90,16 +90,81 @@ describe("GET /api/articles", () => {
             comment_count: expect.any(Number),
           });
         });
-      })
-    });
-    it("responds with a 404 Not Found error for invalid paths", () => {
-      return request(app)
-        .get("/invalid-path")
-        .expect(404)
-        .then((res) => {
-          expect(res.body.message).toBe("Not Found");
-        });
-    });
+      });
   });
+  it("responds with a 404 Not Found error for invalid paths", () => {
+    return request(app)
+      .get("/invalid-path")
+      .expect(404)
+      .then((res) => {
+        expect(res.body.message).toBe("Not Found");
+      });
+  });
+});
 
+describe("GET /api/articles/:article_id/comments", () => {
+  it("responds with an empty array for an article with no comments", () => {
+    const articleId = 2;
+    return request(app)
+      .get(`/api/articles/${articleId}/comments`)
+      .expect(200)
+      .then((res) => {
+        const comments = res.body.comments;
+        expect(comments).toHaveLength(0);
+      });
+  });
+  it("responds with an array of comments for the given article_id with the required properties", () => {
+    let articleId = 1;
+    return request(app)
+      .get(`/api/articles/${articleId}/comments`)
+      .expect(200)
+      .then((res) => {
+        const comments = res.body.comments
+        let commentCount = 0;
+        let i = 0;
+        while (i < testData.commentData.length) {
+          if ((testData.commentData[i].article_id === articleId)) {
+            commentCount++;
+          }
+          i++;
+        }
+        expect(comments.length).toBe(commentCount);
```

Comment:
I enjoy that you've tried to handle this programatically, but it's currently a loop that checks all returned comments have the correct article_id, but not that all comments for a specific article_id have been returned.

You can just hardcode the length value by looking at the test data you have access to (e.g. expect(comments.length).toBe(14))